### PR TITLE
publish kind-projector with full Scala version

### PR DESCRIPTION
**Pull request by sirocchj**
I don't think it makes too much sense to abstract `StringParser` but let me know if you feel otherwise

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -4,7 +4,22 @@ licenses += ("MIT", url("http://opensource.org/licenses/MIT"))
 homepage := Some(url("http://github.com/typelevel/kind-projector"))
 
 scalaVersion := "2.12.8"
-crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.8", "2.13.0")
+crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.8", "2.12.9", "2.12.10", "2.13.0", "2.13.1")
+crossVersion := CrossVersion.full
+crossTarget := {
+  // workarond for https://github.com/sbt/sbt/issues/5097
```
Comment (sirocchj) [inline build.sbt]: credit https://github.com/wartremover/wartremover/commit/5c59c86008b4de9ddc6854a01e4340194dab3ccf#diff-fdc3abdfd754eeb24090dbd90aeec2ceR96

sbt issue https://github.com/sbt/sbt/issues/5097

File: build.sbt [hunk 2]
```diff
@@ -4,7 +4,22 @@ licenses += ("MIT", url("http://opensource.org/licenses/MIT"))
...
+unmanagedSourceDirectories in Compile ++= {
+  (unmanagedSourceDirectories in Compile).value.map { dir =>
+    val sv = scalaVersion.value
+    val is130 = sv == "2.13.0" // use 2.12 version for 2.13.0, reporters changed in 2.13.1
+    CrossVersion.partialVersion(sv) match {
+      case Some((2, n)) if n < 13 || is130 => file(dir.getPath ++ "-2.12-")
+      case _                               => file(dir.getPath ++ "-2.13+")
+    }
+  }
+}
```
Comment (sirocchj) [inline build.sbt]: https://github.com/ghik/silencer/pull/33/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR26-R35

File: build.sbt (hunk 1 above)
Comment (miriarte) [reply build.sbt]: `workaround` maybe?

**Comment by sirocchj** [comment]
Fixes #15 

See also #116 and #117

File: build.sbt [hunk 3]
```diff
@@ -4,7 +4,22 @@ licenses += ("MIT", url("http://opensource.org/licenses/MIT"))
 homepage := Some(url("http://github.com/typelevel/kind-projector"))
 
 scalaVersion := "2.12.8"
-crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.8", "2.13.0")
+crossScalaVersions := Seq("2.10.7", "2.11.12", "2.12.8", "2.12.9", "2.12.10", "2.13.0", "2.13.1")
```
Comment (larsrh) [inline build.sbt]: With the proliferation of Scala versions in here, it makes sense to use https://github.com/dwijnand/sbt-travisci.

**Comment by SethTisue** [comment]
> However, I also have another suggestion how we could avoid changing the cross-versioning scheme. Have we considered using reflection? @SethTisue how would you feel about that?

this recent compatibility breakage hasn't really changed my view expressed at #15. (namely: publishing with full Scala version is certainly more correct, but the status quo isn't terrible either; I have no strong opinion either way)

**Comment by SethTisue** [comment]
thought: maybe @scala-steward could be made smart about how to perform this upgrade, to mitigate the pain downstream?

**Comment by non** [comment]
Hey folks, I've only used sbt-ci-release in one other project (Paiges) but it worked OK once I figured out how to use it, so no objections there. It's a bummer that our daring run of publishing with incomplete Scala versions has to come to an end, but I endorse full cross-versioning. Now that it's happened once it's likely to keep happening, and I'd rather not hack around it and risk future breakage.

**Comment by larsrh** [comment]
> Now that breakage has happened once it's likely to keep happening, and I'd rather not hack around it and risk future breakage.

Fair point. @sirocchj could you make the changes I suggested to the PR, then we can go forward.

File: build.sbt (hunk 3 above)
Comment (sirocchj) [reply build.sbt]: Are you suggesting I add this in this PR?

**Comment by sirocchj** [comment]
One question for you Lars re travis sbt plugin. Will make those changes in about 1h, I’m mobile atm

File: build.sbt (hunk 3 above)
Comment (larsrh) [reply build.sbt]: Yes, that would be great!

**Comment by sirocchj** [comment]
argh
`[error] Switch failed: no subprojects list "2.10.7" (or compatible version) in crossScalaVersions setting.`

**Comment by sirocchj** [comment]
So that was fun 🙄 
the hack to pass the failures [here](https://travis-ci.org/typelevel/kind-projector/builds/591388231) was to explicitly set `crossScalaVersions := (ThisBuild / crossScalaVersions).value`...
I then decided to take to opportunity to "update" the build itself. Let me know if you disagree and I can revert to the hack I had. To me, this now appears much cleaner but YMMV

**Comment by larsrh** [comment]
Sorry for the additional work :pensive: Paging @dwijnand, is that to be expected?

**Comment by sirocchj** [comment]
Not at all :) and I may have done something silly, not sure

**Comment by sirocchj** [comment]
@larsrh should this be moved forward or do you have concerns on the rewrite of the build?

**Comment by larsrh** [comment]
@sirocchj I have this high on my priority list, but I'm at a company event until Thursday and won't have time to look at it. Feel free to ping me again if you don't hear back from me this week.

**Comment by larsrh** [comment]
Thanks much! I'm going ahead with the release now.

**Pull request merged**